server/src/main/java/org/elasticsearch/common/regex/Regex.java

```diff
@@ -48,6 +48,10 @@ public static boolean isSuffixMatchPattern(String str) {
         return str.length() > 1 && str.indexOf('*') == str.length() - 1;
     }
 
+    public static boolean isDottedWildcard(String str) {
+        return isSuffixMatchPattern(str) && str.endsWith(".*");
```

Comment:
Could you add javadoc to this method and `isSuffixMatchPattern`. The name `isSuffixMatchPattern` doesn't scream "this matches a string followed by anything followed by more stuff.

Come to think about it, `isDottedWildcard` might be better named `isSuffixWildcard`. Or something. In my head I call these "prefix matches" so I don't use the word `suffix`. But the rest of the code does. I dunno. grumbling about names.

Reply:
> Could you add javadoc to this method and `isSuffixMatchPattern`. The name `isSuffixMatchPattern` doesn't scream "this matches a string followed by anything followed by more stuff.
> 
> Come to think about it, `isDottedWildcard` might be better named `isSuffixWildcard`. Or something. In my head I call these "prefix matches" so I don't use the word `suffix`. But the rest of the code does. I dunno. grumbling about names.

Will add javadoc for both of them. Thinking about it more, perhaps I'll just use the existing `isSuffixMatchPattern`, instead of adding `isDottedWildcard` .